lumicks/pylake/kymotracker/kymoline.py

```diff
@@ -349,6 +349,25 @@ def _concatenate_lines(self, starting_line, ending_line):
         self._src[self._src.index(starting_line)] = starting_line + ending_line
         self._src.remove(ending_line)
 
+    def _merge_lines(self, starting_line, starting_node, ending_line, ending_node):
+        starting_node = int(starting_node) + 1
+        ending_node = int(ending_node)
+
+        first_half = KymoLine(
+            starting_line.time_idx[:starting_node],
+            starting_line.coordinate_idx[:starting_node],
+            starting_line._image,
+        )
+
+        last_half = KymoLine(
+            ending_line.time_idx[ending_node:],
+            ending_line.coordinate_idx[ending_node:],
+            ending_line._image,
```

Comment:
This is more a workflow question.

Shouldn't there be two, three or four lines (depending on whether you clicked endpoints or not) here since now we are merging in the middle? The mental model that I have of this operation is that basically, what you do is you cut both lines at the point where you click and release, and then connect the new ends tip to tip. Otherwise you risk losing otherwise good traces (see below).

![oops](https://user-images.githubusercontent.com/19836026/154282278-81a1b806-e917-4085-8f37-0e72d91529b6.gif)

This was never an issue when merging only from tip to tip.

Reply:
Second related question is, what if the start and end line are the same? I could imagine what I attempt in the gif below to be useful:
![self_connect](https://user-images.githubusercontent.com/19836026/154287928-feaa299d-14c0-4f73-80e8-8a263ba84172.gif)